src/ai.js

```diff
@@ -18,11 +19,790 @@ function getInternalGame() {
+            else return t;
+        });
+        this.history.push({ type: "selectTarget", data: mapped });
+
+        game.log(`AI: Selected ${mapped[0]}`, !this.logAction);
+
+        if (best_target.length <= 0) {
+            // No targets
+            return false;
+        }
+
+        if (typeof best_target[0] === "number") throw game.functions.createAIError("selectTarget_invalid_return_target", "Card | Player", best_target[0]);
+
+        return best_target[0];
+    }
+
+    /**
+     * Choose the best minion to discover.
+     * 
+     * @param {Card[] | import("./types").Blueprint[]} cards The cards to choose from
+     * 
+     * @returns {Card} Result
+     */
+    discover(cards) {
+        return this._selectFromCards(cards, "discover");
+    }
+
+    /**
+     * Choose the best card to dredge.
+     * 
+     * @param {Card[]} cards The cards to choose from
+     * 
+     * @returns {Card} Result
+     */
+    dredge(cards) {
+        return this._selectFromCards(cards, "dredge");
+    }
+
+    /**
+     * Choose the best option from `options`
+     * 
+     * @param {string[]} options The options the ai can pick from
+     *
+     * @returns {number | null} The question chosen
+     */
+    chooseOne(options) {
+        // TODO: Add this
+        return this.backup_ai.chooseOne(options);
+    }
+
+    /**
+     * Choose the best answer from `options`
+     *
+     * @param {string} prompt The prompt to show to the ai
+     * @param {string[]} options The options the ai can pick from
+     *
+     * @returns {number | null} The index of the option chosen + 1
+     */
+    question(prompt, options) {
+        // TODO: Add this
+        return this.backup_ai.question(prompt, options);
+    }
+
+    /**
+     * Choose yes or no based on the prompt
+     *
+     * @param {string} prompt The prompt to show to the ai
+     *
+     * @returns {boolean} `true` if "Yes", `false` if "No"
+     */
+    yesNoQuestion(prompt) {
+        // TODO: Add this
+        return this.backup_ai.yesNoQuestion(prompt);
+    }
+
+    /**
+     * Returns if the ai wants `card` to be traded
+     *
+     * @param {Card} card The card to check
+     *
+     * @returns {boolean} If the card should be traded
+     */
+    trade(card) {
+        // TODO: Add this
+        return this.backup_ai.trade(card);
+    }
+
     /**
-     * Sentiment-based AI
+     * Returns the list of cards the ai wants to mulligan.
      * 
+     * @returns {string} The indexes of the cards to mulligan. Look in `Interact.mulligan` for more details.
+     */
+    mulligan() {
+        // TODO: Add this
+        return this.backup_ai.mulligan();
+    }
+
+    _selectFromCards(cards, history_name, care_about_mana = true) {
+        // This is currently being used by discover and dredge.
+
+        // This is awful, but this looks like a good template for the other ai methods.
+        // FIXME: This function causes memory leaks.
+        // ^^^^ It might just be horrendus performance.
```

Comment:
Is any of this still true? Remember to remove these types of comments if they no longer apply